src/components/flat-table/flat-table.style.js

```diff
@@ -2,14 +2,42 @@ import styled, { css } from 'styled-components';
 import StyledFlatTableHeader from './flat-table-header/flat-table-header.style';
 import StyledFlatTableRowHeader from './flat-table-row-header/flat-table-row-header.style';
 import StyledFlatTableHead from './flat-table-head/flat-table-head.style';
+import { baseTheme } from '../../style/themes';
 
 const StyledFlatTableWrapper = styled.div`
   height: 100%;
+
+  ${({
+    colorTheme, headerBackground, theme
+  }) => {
+    switch (colorTheme) {
+      case 'dark':
+        return css`
+        ${StyledFlatTableHeader} {
+          background-color: ${theme.flatTable.dark.headerBackground};
+          border-right: 1px solid ${theme.flatTable.dark.border};
+          color: ${theme.colors.white};
+        }`;
+
+      case 'light':
+        return css`
+        ${StyledFlatTableHeader} {
+          background-color: ${theme.flatTable.light.headerBackground};  
+          border-right: 1px solid ${theme.flatTable.light.border};
+        }`;
+
+      default:
+        return css`
+        ${StyledFlatTableHeader} {
+          background-color: ${headerBackground || theme.flatTable.default.headerBackground};
```

Comment:
Could we not `background-color: inherit` here and then add a `case: 'white'` when it becomes necessary